sklearn/compose/_column_transformer.py

```diff
@@ -406,7 +432,7 @@ def _iter(self, fitted, column_as_labels, skip_drop, skip_empty_columns):
             - weight : the weight of the transformer
         """
         if fitted:
-            transformers = self.transformers_
+            transformers = _with_disabled_dtype_warnings(self.transformers_)
```

Comment:
it could be worth just before `if fitted` to add a small comment to mention that we disable the warning for the remainder.